joern-cli/frontends/csharpsrc2cpg/src/main/scala/io/joern/csharpsrc2cpg/astcreation/AstForExpressionsCreator.scala

```diff
@@ -289,12 +292,42 @@ trait AstForExpressionsCreator(implicit withSchemaValidation: ValidationMode) {
   protected def astForSimpleMemberAccessExpression(accessExpr: DotNetNodeInfo): Seq[Ast] = {
     val fieldIdentifierName = nameFromNode(accessExpr)
 
-    val fieldInScope = scope.findFieldInScope(fieldIdentifierName)
-
-    val typeFullName = fieldInScope.map(_.typeFullName).getOrElse(Defines.Any)
-    val identifierName =
-      if (fieldInScope.nonEmpty && fieldInScope.get.isStatic) scope.surroundingTypeDeclFullName.getOrElse(Defines.Any)
-      else "this"
+    val (identifierName, typeFullName) = accessExpr.node match {
+      case SimpleMemberAccessExpression => {
+        createDotNetNodeInfo(accessExpr.json(ParserKeys.Expression)).node match
+          case SuppressNullableWarningExpression =>
+            val baseNode         = createDotNetNodeInfo(accessExpr.json(ParserKeys.Expression)(ParserKeys.Operand))
+            val baseAst          = astForNode(baseNode)
+            val baseTypeFullName = getTypeFullNameFromAstNode(baseAst)
+
+            val fieldInScope = scope.tryResolveFieldAccess(fieldIdentifierName, typeFullName = Option(baseTypeFullName))
+
+            (
+              nameFromNode(baseNode),
+              fieldInScope
+                .map(_.typeName)
+                .getOrElse(Defines.Any)
+            )
+          case _ => {
+            val fieldInScope = scope.findFieldInScope(fieldIdentifierName)
+            val _identifierName =
+              if (fieldInScope.nonEmpty && fieldInScope.get.isStatic)
+                scope.surroundingTypeDeclFullName.getOrElse(Defines.Any)
+              else "this"
```

Comment:
`import io.joern.csharpsrc2cpg.Constants`
```suggestion
              else Constants.This
```

Reply:
(and below)